src/main/java/org/eln2/mc/common/blocks/cell/WireCellBlock.kt

```diff
@@ -1,11 +1,209 @@
+import net.minecraft.world.level.block.state.StateDefinition
+import net.minecraft.world.level.block.state.properties.DirectionProperty
+import net.minecraft.world.level.block.state.properties.BooleanProperty;
+import net.minecraft.world.phys.shapes.CollisionContext
+import net.minecraft.world.phys.shapes.VoxelShape
 import org.eln2.mc.common.blocks.CellBlockBase
+import org.eln2.mc.common.blocks.CellTileEntity
 import org.eln2.mc.common.cell.CellRegistry
 
 class WireCellBlock : CellBlockBase() {
+
+
+
+
+
+
+    init {
+        val connect_east = BooleanProperty.create("connect_east")
+        val connect_north = BooleanProperty.create("connect_north")
+        val connect_south = BooleanProperty.create("connect_south")
+        val connect_west = BooleanProperty.create("connect_west")
+        registerDefaultState(stateDefinition.any().setValue(connect_east, false).setValue(connect_north, false).setValue(connect_south, false).setValue(connect_west, false))
+    }
+
+
     override fun getCellProvider(): ResourceLocation {
         return CellRegistry.WIRE_CELL.id
     }
+
+    override fun getCollisionShape(pState: BlockState, pLevel: BlockGetter, pPos: BlockPos, pContext: CollisionContext): VoxelShape {
+        return Block.box(0.0, 0.0, 0.0, 16.0, 1.0, 16.0)
+    }
+
+    override fun getShape(pState: BlockState, pLevel: BlockGetter, pPos: BlockPos, pContext: CollisionContext): VoxelShape {
+        return Block.box(0.0, 0.0, 0.0, 16.0, 2.0, 16.0)
+    }
+
+    override fun onPlace(pState: BlockState, pLevel: Level, pPos: BlockPos, pOldState: BlockState, pIsMoving: Boolean) {
+
+
+
+        super.onPlace(pState, pLevel, pPos, pOldState, pIsMoving)
+
+
+        // get blocks around and call nechang
+
+        for (x in -1..1) {
+            for (z in -1..1) {
+                if (x == 0 && z == 0) continue
+                neChang(pLevel, pPos, BlockPos(pPos.x + x, pPos.y, pPos.z + z))
+            }
+        }
+
+
+
+
+
+    }
+
+    override fun createBlockStateDefinition(pBuilder: StateDefinition.Builder<Block, BlockState>) {
+        super.createBlockStateDefinition(pBuilder)
+        val connect_east = BooleanProperty.create("connect_east")
+        val connect_north = BooleanProperty.create("connect_north")
+        val connect_south = BooleanProperty.create("connect_south")
+        val connect_west = BooleanProperty.create("connect_west")
+        pBuilder.add(connect_east, connect_north, connect_south, connect_west)
+    }
+
+    fun neChang(level: Level, pos: BlockPos, neighbor: BlockPos) {
+        val connect_east = BooleanProperty.create("connect_east")
+        val connect_north = BooleanProperty.create("connect_north")
+        val connect_south = BooleanProperty.create("connect_south")
+        val connect_west = BooleanProperty.create("connect_west")
+
+
+
+        val blockEntity = level.getBlockEntity(neighbor ?: error("Neighbor position was null"))
+
+        println("Neighbor block entity: $blockEntity")
```

Comment:
I'm only going to comment once for all of these, but we should be using the log4j logger for all logging calls (and not println). We should also make sure that we remove most of these log lines and put them behind the debugger so that we're not constantly spamming the console chat with debug information when we dist the game.

(ditto the whitespace comments for this whole section)

Reply:
Fixed.